datadog/resource_datadog_security_monitoring_default_rule.go

```diff
@@ -256,9 +280,29 @@ func buildSecMonDefaultRuleUpdatePayload(currentState datadogV2.SecurityMonitori
 	}
 	payload.Filters = payloadFilters
 
+	payload.Options = buildDefaultRulePayloadOptions(d)
+
 	return &payload, true, nil
 }
 
+func buildDefaultRulePayloadOptions(d *schema.ResourceData) *datadogV2.SecurityMonitoringRuleOptions {
+	tfOptions := extractMapFromInterface(d.Get("options").([]interface{}))
+	payloadOptions := datadogV2.NewSecurityMonitoringRuleOptions()
+	emptyOptions := true
+	ruleType := d.Get("type").(string)
+
+	if v, ok := tfOptions["decrease_criticality_based_on_env"]; ok && ruleType == string(datadogV2.SECURITYMONITORINGRULETYPECREATE_LOG_DETECTION) {
+		payloadOptions.SetDecreaseCriticalityBasedOnEnv(v.(bool))
+		emptyOptions = false
```

Comment:
Nit: Instead of using `emmptyOptions`, can we just `return payloadOptions` here and `return nil` below?

Reply:
+1 on this. You can just check the length of the `tfOptions` map and return `nil` if it's empty. Also, does the api properly handle a null option or should the option not be set all on empty?